plib/library/IpAddress.php

```diff
@@ -34,8 +34,17 @@ public static function getAvailable()
         throw new pm_Exception("Unable to get server IP: empty result.");
     }
 
+    public static function getPublic($ipAddress)
+    {
+        if (!isset(self::$_ipAddresses)) {
+            self::getAvailable();
+        }
+
+        return self::$_ipAddresses[$ipAddress];
```

Comment:
simplify: `return self::getAvailable()[$ipAddress]` (the method getAvailable is already responsible for the cache)